sbws/core/scanner.py

```diff
@@ -177,10 +177,9 @@ def measure_relay(args, conf, destinations, cb, rl, relay):
     # exit, then pick a non-exit. Otherwise pick an exit.
     helper = None
     circ_fps = None
-    if relay.can_exit_to(dest.hostname, dest.port) and \
-            relay not in rl.bad_exits:
+    if relay not in rl.bad_exits:
         helper = _pick_ideal_second_hop(
-            relay, dest, rl, cb.controller, is_exit=False)
+            relay, dest, rl, cb.controller, is_exit=True)
```

Comment:
With this change, we are now picking a helper that must have the exit flag, and then we proceed to use it as the first hop in a circuit. This is not what you want to be doing here.

Reply:
right, just realized that we ended up measuring in both ways